main_boaz.tex

```diff
@@ -0,0 +1,910 @@
+
+  \item The notation should make it easy to extend operations into tensors with "dangling" axes. For example, if we defined operations on an image or sentence tensor, it should be easy to consider the extension of these operations into a tensor that has an extra batch dimension corresponding to a batch of data points.
+  
+
+\end{itemize}
+
+\subsection{Terminology}
+
+In this section we introduce the terminology that we will use.
+The two main characters we study are \emph{tensors} and \emph{operators} on such tensors.
+In mathematics, we can think of an order $d$ tensor over a field $\mathbb{F}$ as a map from a product set $I_1 \times I_2 \times \cdots \times I_d$ into $\mathbb{F}$. Often these sets $I_1,\ldots,I_d$ are intervals of  integers.
+We will use naming to avoid relying on order for inputs of these tensors.
+We will now define the main concepts we use:
+
+\begin{itemize}
+
+  
+  \item An \emph{axis} is a set of the form $\{ (\text{ax} ,\text{annot},x) | x\in X \}$ where $\text{ax}$ and $\text{annot}$ are strings and $X$ is a well ordered set that can be infinite or finite. (For example, $X$ might be the integers, strings, tuples of integers or strings, or finite subsets of any of those.)   The string $\text{ax}$ is the \emph{name} of the axis, the string $\text{annot}$ is its annotation (can often be the empty string $\text{""}$). An axis with name $\text{ax}$ and empty annotation is denoted by $\name{ax}$. If there is an annotation then we use superscript, subscripts or primes to indicate it. The  set $X$ is the \emph{support} of the axis, and is denoted by $Supp(\name{name})$.  We use the notation $\name{ax} \defaxis X$ to denote that $\name{ax}$ is an axis of the form $\{ (\text{ax},\text{""},x) | x\in  X \}$.  We use the notation $\name{ax}^{annot} \defaxis X$ to denote that $\name{ax}^{\text{annot}}$ is an axis of the form $\{(\text{ax},\text{annot},x) | x\in X \}$. The support of an annotated axis $\name{ax}^{annot}$ is always a subset of the support of the axis $\name{ax}$ with an empty annotation.  If two axes have the same name and annotation then they must have the same support. If $\name{ax}^{annot}$ is an axis and $x \in Supp(X)$, then we denote by $\name{ax}[x]$ the element $(\text{ax},\text{annot},x) \in \name{ax}^{annot}$.  Our default notion for a supporting set will be the natural numbers $\mathbb{N}$ (without zero if we use one-based indexing) and so if we simply say ``''$\name{ax}$ is an ax'' then we assume that its support is the natural numbers, and hence $\name{ax}$ is the set $\{ \name{ax}[1],\name{ax}[2],\ldots \}$.\footnote{This document uses a one-based axising convention, but we can of course do everything with a zero-based axising.}
```

Comment:
If there are axes ax^1 and ax^2, then ax[x] is ambiguous between (ax, 1, x) and (ax, 2, x).

Reply:
There is a typo there which I will fix. Should use ax^{annot}[x] to get (ax,annot,x) 

The way I think about it is ax[x] is (ax,"",x) but it gets automatically casted to (ax,1,x) or (ax,2,x) as needed.

So if a tensor only has one axis of type ax, you can use ax[x] to index it. If there are more than one then you need to use the annotation.